test/jdk/javax/swing/JScrollPane/AcceleratedWheelScrolling/HorizScrollers.java

```diff
@@ -0,0 +1,234 @@
+            Configure Mouse Wheel for Phase %d
+
+            Open the Mouse Control Panel and go to the 'Wheel' tab.
+            If 'Wheel' tab is not available just press Pass.
+
+            %s
+
+            Test the setting on the area below.
+            Once the mouse is setup correctly, the area will turn green.
+            When you're ready for the next part of the test to run, press GO!
+    """;
+
+    static class ConfigPanel extends JPanel
+            implements ActionListener, MouseWheelListener {
+        JTextArea msg;
+        JButton goBtn;
+        JLabel mwArea;
+        int scrollAmount;
+
+        private final Color defaultBg;
+
+        ConfigPanel() {
+            this.scrollAmount = SCROLLAMTS[current];
+            Container content = this;
+            content.setLayout(new BorderLayout());
+            msg = new JTextArea();
+            msg.setMargin(new Insets(5, 5, 5, 5));
+            msg.setEditable(false);
+            msg.setLineWrap(true);
+            msg.setWrapStyleWord(true);
+            content.add(msg, BorderLayout.NORTH);
+
+            mwArea = new JLabel(MW_TEXT, SwingConstants.CENTER);
+            mwArea.setPreferredSize(new Dimension(200, 250));
+            mwArea.setBorder(BorderFactory.createLineBorder(Color.BLACK));
+            mwArea.setOpaque(true);
+            mwArea.addMouseWheelListener(this);
+            content.add(mwArea, BorderLayout.CENTER);
+
+            defaultBg = mwArea.getBackground();
+            setPhase(current);
+
+            goBtn = new JButton("GO!");
+            goBtn.setEnabled(false);
+            goBtn.addActionListener(this);
+            JPanel flowPanel = new JPanel();
+            flowPanel.setLayout(new FlowLayout());
+            flowPanel.add(goBtn);
+            content.add(flowPanel, BorderLayout.SOUTH);
+
+            setPreferredSize(new Dimension(600, 400));
+        }
+
+        public void setPhase(int phase) {
+            if (phase < 3) {
+                setVisible(true);
+                PassFailJFrame.log("Phase %d scroll speed %d"
+                        .formatted(phase + 1, SCROLLAMTS[phase]));
+                PassFailJFrame.log(CONFIG_MSGS[phase]);
+
+                scrollAmount = SCROLLAMTS[phase];
+                msg.setText(CONFIG_INSTRUCTION_TEMPLATE
+                        .formatted(phase + 1, CONFIG_MSGS[phase]));
+                mwArea.setBackground(defaultBg);
+                mwArea.setText(MW_TEXT);
+            } else {
+                // all cases passed
+                showFinalReminderIfNeeded(false);
+            }
+        }
+
+        private void showFinalReminderIfNeeded(boolean isFailure) {
+            if (scrollAmount != 3) {
+                JOptionPane.showMessageDialog(
+                        ConfigPanel.this.getTopLevelAncestor(),
+                        ("Test %s. please make sure you have restored " +
+                                "the original speed value blah blah")
```

Comment:
Do we really need this exact wording?

Reply:
I fixed this as well in the new PR linked in the other comment. Thanks!